python/tests/transformers/tf_tensor_test.py

```diff
@@ -0,0 +1,323 @@
+                            dtype=tf.float64, name='varW')
+            z = tf.reduce_mean(x * w, axis=1, name='tnsrOut')
+            sess.run(w.initializer)
+            saver = tf.train.Saver(var_list=[w])
+            _ = saver.save(sess, ckpt_path_prefix, global_step=2702)
+
+            # Get the reference data
+            _results = []
+            for row in df.collect():
+                arr = np.array(row.vec)[np.newaxis, :]
+                _results.append(sess.run(z, {x: arr}))
+            out_ref = np.hstack(_results)
+
+        transformer = TFTransformer(
+            tfInputGraph=TFInputGraphBuilder.fromCheckpoint(model_ckpt_dir),
+            inputMapping={
+                input_col: 'tnsrIn'
+            },
+            outputMapping={
+                'tnsrOut': output_col
+            })
+        final_df = transformer.transform(analyzed_df)
+        out_tgt = grab_df_arr(final_df, output_col)
+
+        shutil.rmtree(model_ckpt_dir, ignore_errors=True)
+        self.assertTrue(np.allclose(out_ref, out_tgt))
+
+
+    def test_multi_io(self):
+        # Build a simple input DataFrame
+        vec_size = 17
+        num_vecs = 31
+        _df = self._get_rand_vec_df(num_vecs, vec_size)
+        df_x = _df.withColumnRenamed('vec', 'vec_x')
+        _df = self._get_rand_vec_df(num_vecs, vec_size)
+        df_y = _df.withColumnRenamed('vec', 'vec_y')
+        df = df_x.join(df_y, on='idx', how='inner')
+        analyzed_df = tfs.analyze(df)
+
+        # Build the TensorFlow graph
+        with tf.Session() as sess:
+            x = tfs.block(analyzed_df, 'vec_x')
+            y = tfs.block(analyzed_df, 'vec_y')
+            p = tf.reduce_mean(x + y, axis=1)
+            q = tf.reduce_mean(x - y, axis=1)
+            graph = sess.graph
+
+            # Get the reference data
+            p_out_ref = []
+            q_out_ref = []
+            for row in df.collect():
+                arr_x = np.array(row['vec_x'])[np.newaxis, :]
+                arr_y = np.array(row['vec_y'])[np.newaxis, :]
+                p_val, q_val = sess.run([p, q], {x: arr_x, y: arr_y})
+                p_out_ref.append(p_val)
+                q_out_ref.append(q_val)
+            p_out_ref = np.hstack(p_out_ref)
+            q_out_ref = np.hstack(q_out_ref)
+
+        # Apply the transform
+        transfomer = TFTransformer(
+            tfInputGraph=TFInputGraphBuilder.fromGraph(graph),
+            inputMapping={
+                'vec_x': x,
+                'vec_y': y
+            },
+            outputMapping={
+                p: 'outcol_p',
+                q: 'outcol_q'
+            })
+        final_df = transfomer.transform(analyzed_df)
+        p_out_tgt = grab_df_arr(final_df, 'outcol_p')
+        q_out_tgt = grab_df_arr(final_df, 'outcol_q')
+
+        self.assertTrue(np.allclose(p_out_ref, p_out_tgt))
+        self.assertTrue(np.allclose(q_out_ref, q_out_tgt))
+
+    def test_mixed_keras_graph(self):
```

Comment:
what are we trying to accomplish with this test? is the goal to test a larger graph (doesn't look that big?)? or test compatibility with keras?

Reply:
The motivation is to see if we can build TFTransformer from TF + Keras mixed layers.

Reply:
It's good to know that works, though I am not sure why we would question that it works in this context (i.e. this doesn't seem like the right place to test that vs tensorframes and/or the strip_and_freeze unit tests - yes i know we don't have any). in any case, i'd leave it in for now but add a TODO comment to remove it once we add the keras transformer.